### DISCUSSION TITLE
How to load .env file into the test environment?

### DISCUSSION THREAD

**Discussion by datsenkoboos** [vitest-dev/vitest#3989] [Q&A]
I want to use variables from **.env.public** file in my test environment, so i've added `setupFiles: '.env.public'` to **vitest.config.ts**:

![изображение](https://github.com/vitest-dev/vitest/assets/88643277/d2c7d142-4962-4708-a37e-2ee16a3f0c2a)

**.env.public** and **vite.config.ts** are both in the root of my project:

![изображение](https://github.com/vitest-dev/vitest/assets/88643277/ff776355-8604-4c6e-9c03-7ca912ee104f)

Tests are failing with error:

![изображение](https://github.com/vitest-dev/vitest/assets/88643277/efc838b0-6a19-447c-84e0-a5c6a4e41572)

**ctrl + click** on the url from the error successfully opens **.env.public** file

How can i fix the issue?

**Comment by keymatch-clovis** [comment]
If anyone still stumbles into this problem, I have done the following:

I'm using the [`loadEnvFile`](https://nodejs.org/docs/v22.13.1/api/process.html#processloadenvfilepath) function from Node.js. Keep in mind this was added for versions v21.7.0, v20.12.0. If you have an older version, or don't want to use this as it's still under active development, you could use a library like [dotenv](https://www.npmjs.com/package/dotenv), if I understand correctly.

I'm using the Vitest configuration [`setupFiles`](https://vitest.dev/config/#setupfiles) property, something like:

```typescript
import swc from "unplugin-swc";
import { defineConfig } from "vitest/config";

export default defineConfig({
    // ...
    test: {
        // ...
        setupFiles: ["./test/setup.ts"],
    },
    // ...
});
```

And in the `./test/setup.ts` file, I add:

```typescript
import { loadEnvFile } from "node:process";

// We need to load the local .env file to access some variables needed when
// testing.
loadEnvFile();
```

I understand you could set the `path` argument for the `loadEnvFile` to use other file, and not the default `'./.env'` file.
